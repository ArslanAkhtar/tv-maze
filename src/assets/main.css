.v-main{
  background-color: black;
  min-height: 90vh;
}


.movie-summary {
  display: -webkit-box;
  -webkit-line-clamp: 3;
  -webkit-box-orient: vertical;
  overflow: hidden;
}

.genre {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: flex-start;
}

.loading {
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
}

.show-detail {
  display: flex;
  align-items: baseline;
  gap: 5px;
  margin: 10px 0;
}

.loadMore {
  width: 100%;
  display: flex;
  justify-content: center;
}

.no-data-found {
  width: 100%;
  display: flex;
  justify-content: center;
  flex-direction: column;
  align-items: center;
  margin-bottom: 40px;
}

.page-not-found {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex-wrap: wrap;
  justify-content: center;
  height: 100%;
  
}