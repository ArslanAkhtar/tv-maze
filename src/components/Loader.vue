<script setup lang="ts">
// import { FulfillingBouncingCircleSpinner } from 'epic-spinners'
</script>
<template>
  <div class="loading">
    <v-progress-circular indeterminate model-value="20" :size="51" :width="7"></v-progress-circular>
  </div>
</template>
