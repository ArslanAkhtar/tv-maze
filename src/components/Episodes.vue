<script setup lang="ts">
	import useFetchShows from "@/composables/FetchData";
	import Loader from "@/components/Loader.vue";
	import { imgURL, openExternalURL } from "@/helpers/utility";
	import Error from "@/components/Error.vue";

	const props = defineProps({
		id: {
			type: Number,
			required: true,
		},
	});

	const { loadShowEpisodes } = useFetchShows();
	const {
		data: episodes,
		isFetching,
		error: ErrorText,
	} = loadShowEpisodes(props.id);
</script>
<template>
	<v-list :height="300">
		<v-list-item
			v-if="episodes"
			v-for="(episode, i) in episodes"
			:key="i"
			@click="() => openExternalURL(episode.url)"
			:prepend-avatar="imgURL(episode.image?.medium)"
			class="pa-5"
			elevation="1"
		>
			<template v-if="episode.name" v-slot:title>
				{{ episode.name }}
			</template>
			<template v-if="episode.number" v-slot:subtitle>
				Episode {{ episode.number }}
			</template>
			<template v-slot:append>
				<v-icon>mdi-play</v-icon>
			</template>
		</v-list-item>
		<Loader v-else-if="isFetching" />
		<div class="d-flex justify-center align-center" style="height: 100%" v-else>
			<Error :text="ErrorText" />
		</div>
	</v-list>
</template>
