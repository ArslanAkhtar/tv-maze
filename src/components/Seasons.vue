<script setup lang="ts">
	import Episodes from "@/components/Episodes.vue";
	import type { Season } from "@/helpers/types";
	import type { PropType } from "vue";

	defineProps({
		seasons: {
			type: Array as PropType<Season[]>,
			required: true,
		},
	});
</script>

<template>
	<div class="seasons" v-if="seasons">
		<div class="text-h5 mb-5 font-weight-bold">Seasons:</div>
		<v-expansion-panels>
			<v-expansion-panel
				v-for="season in seasons"
				:key="season.id"
				:title="'Season ' + season.number"
				expand-icon="mdi-chevron-down"
				collapse-icon="mdi-chevron-up"
			>
				<v-expansion-panel-text>
					<Episodes :id="season.id" />
				</v-expansion-panel-text>
			</v-expansion-panel>
		</v-expansion-panels>
	</div>
</template>
