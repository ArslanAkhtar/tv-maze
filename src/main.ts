import './assets/main.css'

import { createApp } from 'vue'
import App from './App.vue'
import router from './router'
import 'vuetify/styles'
import { createVuetify } from 'vuetify'
import * as components from 'vuetify/components'
import * as directives from 'vuetify/directives'
import { fa } from 'vuetify/iconsets/fa'
import { mdi, aliases } from 'vuetify/iconsets/mdi'

import '@mdi/font/css/materialdesignicons.css'

const app = createApp(App)

app.use(router)

const vuetify = createVuetify({
  components,
  directives,
  theme: {
    defaultTheme: 'dark'
  },
  icons: {
    defaultSet: 'mdi',
    aliases,
    sets: {
      mdi,
      fa
    }
  }
})

app.use(vuetify)

app.mount('#app')
