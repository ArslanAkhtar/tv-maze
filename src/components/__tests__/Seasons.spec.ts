import { describe, it, expect } from "vitest";
import useFetchShows from "@/composables/FetchData";
import Seasons from "../Seasons.vue";
import { mount, shallowMount } from "@vue/test-utils";
import { vuetify } from "@/main";
import mockSeasons from "./mockData/mockSeason.json";
import Episodes from "../Episodes.vue";
import Loader from "../Loader.vue";

describe("Seasons Component", () => {
	const wrapper = mount(Seasons, {
		props: { seasons: mockSeasons },
		global: {
			components: {
				Episodes,
				Loader,
			},
		},
		plugins: [vuetify],
	});

	test("renders correctly", () => {
		expect(wrapper.element).toMatchSnapshot();
	});

	it("renders props", () => {
		expect(wrapper.props().seasons).toMatchObject(mockSeasons);
	});
});
