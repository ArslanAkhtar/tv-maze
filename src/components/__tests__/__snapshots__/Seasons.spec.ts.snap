// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Seasons Component > renders correctly 1`] = `
<div
  class="seasons"
>
  <div
    class="text-h5 mb-5 font-weight-bold"
  >
    Seasons:
  </div>
  <v-expansion-panels>
    
    <v-expansion-panel
      collapse-icon="mdi-chevron-up"
      expand-icon="mdi-chevron-down"
      title="Season 1"
    >
      <v-expansion-panel-text>
        <v-list
          height="300"
        >
          <div
            class="loading"
          >
            <v-progress-circular
              indeterminate=""
              model-value="20"
              size="51"
              width="7"
            />
          </div>
        </v-list>
      </v-expansion-panel-text>
    </v-expansion-panel>
    <v-expansion-panel
      collapse-icon="mdi-chevron-up"
      expand-icon="mdi-chevron-down"
      title="Season 2"
    >
      <v-expansion-panel-text>
        <v-list
          height="300"
        >
          <div
            class="loading"
          >
            <v-progress-circular
              indeterminate=""
              model-value="20"
              size="51"
              width="7"
            />
          </div>
        </v-list>
      </v-expansion-panel-text>
    </v-expansion-panel>
    <v-expansion-panel
      collapse-icon="mdi-chevron-up"
      expand-icon="mdi-chevron-down"
      title="Season 3"
    >
      <v-expansion-panel-text>
        <v-list
          height="300"
        >
          <div
            class="loading"
          >
            <v-progress-circular
              indeterminate=""
              model-value="20"
              size="51"
              width="7"
            />
          </div>
        </v-list>
      </v-expansion-panel-text>
    </v-expansion-panel>
    
  </v-expansion-panels>
</div>
`;
